source/robot_model/src/Model.cpp

```diff
@@ -184,6 +184,34 @@ bool Model::check_collision(const state_representation::JointPositions& joint_po
   return false;
 }
 
+Eigen::MatrixXd Model::compute_minimum_distance(const state_representation::JointPositions& joint_positions) {
+  if (!this->is_geometry_model_initialized()) {
+    throw robot_model::exceptions::CollisionGeometryException(
+        "Geometry model not loaded for " + this->get_robot_name());
+  }
+  Eigen::VectorXd configuration = joint_positions.get_positions();
+  pinocchio::computeDistances(
+      this->robot_model_, this->robot_data_, this->geom_model_, this->geom_data_, configuration);
+
+  // nb_joints is the number of joints in the robot model
+  unsigned int nb_joints = this->get_number_of_joints();
+
+  // create a 2D Eigen vector to store the distances and initialize to zero
```

Comment:
nitpick: "2D vector" generally implies a 1D tensor with two elements (i.e. `Vector2d`). You mean a 2D tensor, which we normally just call matrix

```suggestion
  // create a square matrix to store the distances and initialize to zero
```